engine/grid.py

```diff
@@ -393,6 +399,8 @@ class Direction(Enum):
         while (phase != WaypointPhase.TERMINATE):
             if direction == Direction.LEFT:
                 new_pos = (curr_pos[0]-1,curr_pos[1])
+                if not self.is_position_in_grid_bounds(new_pos):
```

Comment:
currently, this break statement seems to terminate the traversal prematurely, so the entire grid is not traversed. I tested with a rectangle traversal with ` grid.activate_rectangle(0, 0, rows, cols)`

Reply:
oh yes! that makes sense I totally missed the purple line o7